### ISSUE TITLE
Can't update OSM data with 3.3

### ISSUE THREAD

**Issue by maybephil** [mediagis/nominatim-docker#66]
Hi,

I am using the latest 3.3 Dockerfile. After a fresh build and importing the data I want to setup the update mechanism like it is described here: https://github.com/mediagis/nominatim-docker/blob/master/3.3/README.md#update

```
→ sudo docker exec -it nominatim sudo -u postgres ./src/build/utils/update.php --init-updates
2019-05-24 12:14:07 == module path: /app/src/build/module
2019-05-24 12:14:07 == Create Functions
ERROR: HTTP Error 404: Not Found
ERROR: HTTP Error 404: Not Found
ERROR: HTTP Error 404: Not Found
ERROR: HTTP Error 404: Not Found
ERROR: HTTP Error 404: Not Found
Done. Database updates will start at sequence 2252 (2019-05-18T17:12:29Z)
```

Seems fine, although I did not look into 404er errors.

```
→ sudo docker exec -it nominatim sudo -u postgres ./src/build/utils/update.php --import-osmosis-all
Currently at sequence 2252 (18/05/2019 20:12:29) - 1 indexed
/usr/local/bin/pyosmium-get-changes --server https://download.geofabrik.de/europe-updates -o /app/src/build/osmosischange.osc -s 30 -I 2252
Traceback (most recent call last):
  File "/usr/local/bin/pyosmium-get-changes", line 243, in <module>
    exit(main(sys.argv[1:]))
  File "/usr/local/bin/pyosmium-get-changes", line 224, in main
    outhandler = WriteHandler(options.outfile)
RuntimeError: Open failed for '/app/src/build/osmosischange.osc': Permission denied
ERROR: updates failed.%
```

This is the result when I run the update script. Seems like there is an issue with folder permissions. After manually fixing that and adding in the container I get the following message:

```
→ sudo docker exec -it nominatim sudo -u postgres ./src/build/utils/update.php --import-osmosis-all
Currently at sequence 2252 (18/05/2019 20:12:29) - 1 indexed
/usr/local/bin/pyosmium-get-changes --server https://download.geofabrik.de/europe-updates -o /app/src/build/osmosischange.osc -s 30 -I 2252
Traceback (most recent call last):
  File "/app/src/utils/osm_file_date.py", line 3, in <module>
    import osmium
ImportError: No module named osmium
ERROR: Error getting date from diff file.
string(34) "Error getting date from diff file."

```

So it seems like there are modules etc. in the docker image missing to run the update.

Thanks

**Comment by grachevko** [comment]
Same issue

**Comment by maxze** [comment]
The issue here seems to be that the Pyosmium binary cannot be found. Run the command 

`whereis pyosmium-get-changes` 

inside the container and see what it shows. 

Now go into your `locals.php` file and make sure the line

`@define('CONST_Pyosmium_Binary', '/usr/local/bin/pyosmium-get-changes');`

shows the correct path. I remember having that problem some time ago.

**Comment by therek** [comment]
I had the same problem and it turned out that `pyosmium-get-changes` uses `/usr/bin/python3` while Python scripts supplied with Nominatim refer to `/usr/bin/python` which in `ubuntu:disco` image is Python 2.7. Also, user `postgres` have no write permission to /app/src/build.

Here's the patch for Dockerfile I've used to workaround it.
```
--- 3.3/Dockerfile.orig	2019-05-28 13:52:56.347074000 +0200
+++ 3.3/Dockerfile	2019-06-05 16:19:29.442759114 +0200
@@ -40,6 +40,8 @@

 # Load initial data
 RUN curl http://www.nominatim.org/data/country_grid.sql.gz > /app/src/data/country_osm_grid.sql.gz
+RUN sed -i -e 's|bin/python|bin/python3|' /app/src/utils/*.py
+RUN chmod o=rwx /app/src/build

 EXPOSE 5432
 EXPOSE 8080
```

You can apply it with `patch -p0 < patch-Docker`.

**Comment by Wilfli** [comment]
I have the same issue. I have two docker container with postgres and apache but when I trying update db using command ``` docker exec -it nominatim sudo -u postgres ./src/build/utils/update.php --import-osmosis-all ``` I got  
``` 
Currently at sequence 967 (04/07/2019 20:08:23) - 1 indexed
/usr/local/bin/pyosmium-get-changes --server http://download.geofabrik.de/europe/russia-european-part-updates -o /app/src/build/osmosischange.osc -s 30 -I 967
Traceback (most recent call last):
  File "/usr/local/bin/pyosmium-get-changes", line 243, in <module>
    exit(main(sys.argv[1:]))
  File "/usr/local/bin/pyosmium-get-changes", line 224, in main
    outhandler = WriteHandler(options.outfile)
RuntimeError: Open failed for '/app/src/build/osmosischange.osc': Permission denied
```
CONST_Pyosmium_Binary have currect path. Someone can help me with this issue?

**Comment by therek** [comment]
@Wilfli you can use the patch I've supplied above. There's also a pull request #71 with that change.

**Comment by Wilfli** [comment]
Yes it work. Thanks

**Comment by ErwanTremiot** [comment]
It seems that the PR #71 from @therek has been merged, but it only provides a fix for version 3.3 and not for 3.4 version that has the same problem.
I tested the solution from @theHellDog on the 3.4 version and it worked for me:
```Dockerfile
# Fix python default version
RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 3
RUN update-alternatives --install /usr/bin/python python /usr/bin/python2 2
```

**Comment by philipkozeny** [comment]
@ErwanTremiot where exactly do you see the problem? I have built 3.4 with no changes regarding that issue and everything works as expected.
